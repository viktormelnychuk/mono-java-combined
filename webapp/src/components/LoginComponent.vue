<script>
import { AuthService } from "@/services/authService";

export default {
  data() {
    return {
      credentials: {
        username: "",
        password: "",
      },
      loading: false,
      rules: {
        username: [
          {
            required: true,
            message: "Username is required",
            trigger: "blur",
          },
          {
            min: 2,
            message: "Username length should be at least 3 characters",
            trigger: "blur",
          },
        ],
        password: [
          { required: true, message: "Password is required", trigger: "blur" },
          {
            min: 1,
            message: "Password length should be at least 5 characters",
            trigger: "blur",
          },
        ],
      },
    };
  },
  setup() {
    return {
      authService: new AuthService(),
    };
  },
  methods: {
    async submit() {
      const valid = await this.$refs.form.validate();
      if (!valid) {
        return;
      }
      this.loading = true;
      await this.authService.login(this.credentials);
      this.loading = false;
      this.$message.success("Login successful");
      await this.$router.push("/");
      await this.$router.go(0);
    },
  },
};
</script>

<template>
  <div class="login">
    <el-card>
      <h2>Login</h2>
      <el-form
        class="login-form"
        :model="credentials"
        :rules="rules"
        ref="form"
        @submit.prevent="submit"
      >
        <el-form-item prop="username">
          <el-input
            v-model="credentials.username"
            placeholder="Username"
          ></el-input>
        </el-form-item>
        <el-form-item prop="password">
          <el-input
            v-model="credentials.password"
            placeholder="Password"
            type="password"
          ></el-input>
        </el-form-item>
        <el-form-item>
          <el-button
            :loading="loading"
            class="login-button"
            type="primary"
            native-type="submit"
            block
            >Login
          </el-button>
        </el-form-item>
      </el-form>
    </el-card>
  </div>
</template>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
.login {
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
}

.login-button {
  width: 100%;
  margin-top: 40px;
}

.login-form {
  width: 290px;
}

.forgot-password {
  margin-top: 10px;
}

$teal: rgb(0, 124, 137);
.el-button--primary {
  background: $teal;
  border-color: $teal;

  &:hover,
  &.active,
  &:focus {
    background: lighten($teal, 7);
    border-color: lighten($teal, 7);
  }
}

.login .el-input__inner:hover {
  border-color: $teal;
}

.login .el-input__prefix {
  background: rgb(238, 237, 234);
  left: 0;
  height: calc(100% - 2px);
  top: 1px;
  border-radius: 3px;

  .el-input__icon {
    width: 30px;
  }
}

.login .el-input input {
  padding-left: 35px;
}

.login .el-card {
  padding-top: 0;
  padding-bottom: 30px;
}

h2 {
  font-family: "Open Sans", serif;
  letter-spacing: 1px;
  padding-bottom: 20px;
}

a {
  color: $teal;
  text-decoration: none;

  &:hover,
  &:active,
  &:focus {
    color: lighten($teal, 7);
  }
}

.login .el-card {
  width: 340px;
  display: flex;
  justify-content: center;
}
</style>
