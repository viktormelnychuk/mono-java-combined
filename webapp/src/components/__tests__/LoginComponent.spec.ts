import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import LoginComponent from "../LoginComponent.vue";

describe("LoginComponent", () => {
  it("should render", () => {
    const wrapper = mount(LoginComponent);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should shouw error if username is empty", () => {
    const wrapper = mount(LoginComponent);
    wrapper.find("el-input[placeholder='Username']").setValue("any");
    wrapper.find("el-button").trigger('click');
    expect(wrapper).toContain("username is required")
  });
});
