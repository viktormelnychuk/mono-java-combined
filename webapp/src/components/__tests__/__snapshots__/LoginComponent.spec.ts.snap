// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginComponent > should render 1`] = `
"<div data-v-87276d9b=\\"\\" class=\\"login\\">
  <el-card data-v-87276d9b=\\"\\">
    <h2 data-v-87276d9b=\\"\\">Login</h2>
    <el-form data-v-87276d9b=\\"\\" class=\\"login-form\\" model=\\"[object Object]\\" rules=\\"[object Object]\\">
      <el-form-item data-v-87276d9b=\\"\\" prop=\\"username\\">
        <el-input data-v-87276d9b=\\"\\" modelvalue=\\"\\" placeholder=\\"Username\\"></el-input>
      </el-form-item>
      <el-form-item data-v-87276d9b=\\"\\" prop=\\"password\\">
        <el-input data-v-87276d9b=\\"\\" modelvalue=\\"\\" placeholder=\\"Password\\" type=\\"password\\"></el-input>
      </el-form-item>
      <el-form-item data-v-87276d9b=\\"\\">
        <el-button data-v-87276d9b=\\"\\" loading=\\"false\\" class=\\"login-button\\" type=\\"primary\\" native-type=\\"submit\\" block=\\"\\">Login </el-button>
      </el-form-item>
    </el-form>
  </el-card>
</div>"
`;
